File: libgdx_slam_the_bird/core/src/com/nopalsoft/slamthebird/scene2d/DialogRate.java
package com.nopalsoft.slamthebird.scene2d;

import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.ui.Label;
import com.badlogic.gdx.scenes.scene2d.ui.Table;
import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
import com.nopalsoft.slamthebird.Assets;
import com.nopalsoft.slamthebird.screens.BaseScreen;

public class DialogRate extends Dialog {

    public DialogRate(BaseScreen currentScreen) {
        super(currentScreen);
        setSize(390, 260);
        setY(300);
        setBackGround();

        Label labelTitle = new Label("Support this game", Assets.smallLabelStyle);
        labelTitle.setPosition(getWidth() / 2f - labelTitle.getWidth() / 2f, 210);

        Label labelDescription = new Label(
                "Hello, thank you for playing Slam the Bird.\nHelp us to support this game. Just rate us at the app store.",
                Assets.smallLabelStyle);
        labelDescription.setSize(getWidth() - 20, 170);
        labelDescription.setPosition(getWidth() / 2f - labelDescription.getWidth() / 2f,
                50);
        labelDescription.setWrap(true);

        TextButton buttonRate = new TextButton("Rate", Assets.styleTextButtonPurchased);
        screen.addPressEffect(buttonRate);
        buttonRate.getLabel().setWrap(true);
        buttonRate.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                hide();
            }
        });

        TextButton buttonNotNow = new TextButton("Not now", Assets.styleTextButtonSelected);
        screen.addPressEffect(buttonNotNow);
        buttonNotNow.getLabel().setWrap(true);
        buttonNotNow.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                hide();
            }
        });

        Table tableButtons = new Table();
        tableButtons.setSize(getWidth() - 20, 60);
        tableButtons.setPosition(getWidth() / 2f - tableButtons.getWidth() / 2f, 10);

        tableButtons.defaults().uniform().expand().center().fill().pad(10);
        tableButtons.add(buttonRate);
        tableButtons.add(buttonNotNow);

        addActor(labelDescription);
        addActor(tableButtons);
        addActor(labelTitle);
    }
}

File: libgdx_slam_the_bird/core/src/com/nopalsoft/slamthebird/scene2d/DialogRate.kt
package com.nopalsoft.slamthebird.scene2d

import com.badlogic.gdx.scenes.scene2d.InputEvent
import com.badlogic.gdx.scenes.scene2d.ui.Label
import com.badlogic.gdx.scenes.scene2d.ui.Table
import com.badlogic.gdx.scenes.scene2d.ui.TextButton
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener
import com.nopalsoft.slamthebird.Assets
import com.nopalsoft.slamthebird.screens.BaseScreen

class DialogRate(currentScreen: BaseScreen) : Dialog(currentScreen) {
    init {
        setSize(390f, 260f)
        y = 300f
        setBackGround()

        val labelTitle = Label("Support this game", Assets.smallLabelStyle)
        labelTitle.setPosition(width / 2f - labelTitle.width / 2f, 210f)

        val labelDescription = Label(
            "Hello, thank you for playing Slam the Bird.\nHelp us to support this game. Just rate us at the app store.",
            Assets.smallLabelStyle
        )
        labelDescription.setSize(width - 20, 170f)
        labelDescription.setPosition(
            width / 2f - labelDescription.width / 2f,
            50f
        )
        labelDescription.wrap = true

        val buttonRate = TextButton("Rate", Assets.styleTextButtonPurchased)
        screen.addPressEffect(buttonRate)
        buttonRate.label.wrap = true
        buttonRate.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                hide()
            }
        })

        val buttonNotNow = TextButton("Not now", Assets.styleTextButtonSelected)
        screen.addPressEffect(buttonNotNow)
        buttonNotNow.label.wrap = true
        buttonNotNow.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                hide()
            }
        })

        val tableButtons = Table()
        tableButtons.setSize(width - 20, 60f)
        tableButtons.setPosition(width / 2f - tableButtons.width / 2f, 10f)

        tableButtons.defaults().uniform().expand().center().fill().pad(10f)
        tableButtons.add(buttonRate)
        tableButtons.add(buttonNotNow)

        addActor(labelDescription)
        addActor(tableButtons)
        addActor(labelTitle)
    }
}
